kbn_pm/src/lib/bazel.mjs

```diff
@@ -141,11 +141,38 @@ export async function cleanDiskCache(log) {
   log.success('removed disk caches');
 }
 
+/**
+ * @param {string | undefined} version
+ */
+function isYarnV1(version) {
+  if (!version) {
+    return false;
+  }
+
+  // yarn version can be semver so trim any special characters
+  const majorVersion = version.replace(/[/^>=<~*]/, '').split('.')[0];
+
+  return parseInt(majorVersion) < 2;
+}
+
 /**
  * @param {import('./log.mjs').Log} log
  * @param {{ offline?: boolean, quiet?: boolean } | undefined} opts
  */
 export async function installYarnDeps(log, opts = undefined) {
+  const path = Path.resolve(REPO_ROOT, 'package.json');
+  /** @type {import('@kbn/repo-info').KibanaPackageJson} */
+  const pkgJson = JSON.parse(await Fsp.readFile(path, 'utf8'));
+  const sharpVersion = pkgJson.devDependencies?.sharp;
+  const yarnVersion = pkgJson.engines?.yarn;
+
+  // yarn v1 requires --ignore-engines flag for sharp to be installed cross-platform
+  if (isYarnV1(yarnVersion) && sharpVersion) {
```

Comment:
--ignore-engines only apply for sharp >=0.33.0 (which we are currently not using). I think we should only run this logic to accommodate  for future problems